_inc/client/components/masthead/style.scss

```diff
@@ -32,10 +32,9 @@
 	}
 }
 
-.jp-masthead__logo {
+.jetpack-logo__masthead {
```

Comment:
Remove this whole rule set and it will be the right size. :)

Reply:
haha nice

Reply:
@MichaelArestad When I remove this rule, the logo is no longer visible

Reply:
Figured it out in 4ab5ed3